src/bytes.rs

```diff
@@ -2260,6 +2345,94 @@ unsafe impl Sync for Inner {}
  *
  */
 
+impl Inner2 {
+    fn into_vec(self) -> Vec<u8> {
+        let kind = self.inner.kind();
+        if kind == KIND_VEC {
+            let r = unsafe { Vec::from_raw_parts(self.inner.ptr, self.inner.len, self.inner.cap) };
+
+            // no need to call the destructor, we took the data
+            mem::forget(self);
+
+            return r;
+        }
+
+        if kind == KIND_ARC {
+            let arc = self.arc.load(Relaxed);
```

Comment:
Could you add a comment stating that this `Relaxed` load is safe as we are guaranteed to have exclusive access to the atomic by virtue of the function taking `self`.